server/webapp/WEB-INF/rails/webpack/views/pages/config_repos/config_repo_view_model.ts

```diff
@@ -14,78 +14,126 @@
  * limitations under the License.
  */
 
+import {ApiResult} from "helpers/api_request_builder";
 import * as _ from "lodash";
-import {ConfigRepo, humanizedMaterialAttributeName} from "models/config_repos/types";
+import * as m from "mithril";
+import * as stream from "mithril/stream";
+import {Stream} from "mithril/stream";
+import {AbstractObjCache, ObjectCache, rejectAsString} from "models/base/cache";
+import {ConfigReposCRUD} from "models/config_repos/config_repos_crud";
+import {DefinedStructures} from "models/config_repos/defined_structures";
+import {ConfigRepo} from "models/config_repos/types";
 import {EventAware} from "models/mixins/event_aware";
-import {PluginInfo} from "models/shared/plugin_infos_new/plugin_info";
-import {DeleteOperation, EditOperation, RefreshOperation} from "views/pages/page_operations";
+import {DeleteConfirmModal} from "views/components/modal/delete_confirm_modal";
+import {EditConfigRepoModal} from "views/pages/config_repos/modals";
+import {FlashContainer, RequiresPluginInfos, SaveOperation} from "views/pages/page_operations";
 
-interface GenericOperations<T> extends EditOperation<T>, DeleteOperation<T>, RefreshOperation<T> {}
+interface PageResources extends SaveOperation, RequiresPluginInfos, FlashContainer {}
 
-export interface Operations extends GenericOperations<ConfigRepo> {}
+class CRResultCache extends AbstractObjCache<DefinedStructures> {
+  private repoId: string;
+  private etag: Stream<string> = stream();
 
-export interface CRVMAware {
-  vm: CRWidgetVM;
+  constructor(repoId: string) {
+    super();
+    this.repoId = repoId;
+  }
+
+  doFetch(resolve: (data: DefinedStructures) => void, reject: (reason: string) => void) {
+    DefinedStructures.fetch(this.repoId, this.etag()).then((result) => {
+      if (304 === result.getStatusCode()) {
+        resolve(this.contents()); // no change
+        return;
+      }
+
+      if (result.getEtag()) {
+        this.etag(result.getEtag()!);
+      }
+
+      result.do((resp) => {
+        resolve(DefinedStructures.fromJSON(JSON.parse(resp.body)));
+      }, (error) => {
+        reject(error.message);
+      });
+    }).catch(rejectAsString(reject));
+  }
+
+  empty() {
+    // don't dump contents, just force a fresh set of data
+    // this.etag = stream();
```

Comment:
Did you mean to comment this?

Reply:
That's a mistake -- it's not meant to be commented out.